CMakeLists.txt

```diff
@@ -76,43 +76,26 @@ FetchContent_Declare(
 ################################
 # NLOHMANN JSON
 ################################
+# Use URL to avoid downloading large repository as recommended here:
+# https://json.nlohmann.me/integration/cmake/#fetchcontent
 FetchContent_Declare(
         nlohmann-json
-        GIT_REPOSITORY https://github.com/nlohmann/json.git
-        GIT_TAG 9cca280a4d0ccf0c08f47a99aa71d1b0e52f8d03 # release-3.11.3
+        URL https://github.com/nlohmann/json/releases/download/v3.11.3/json.tar.xz
+        URL_HASH SHA3_224=b9237e8e8242e915de5be631aa944369c38a377c5e79aa505ba741a9
 )
-# We only want to include a single file from the nlohmann GitHub repository, so we initialize the submodule
-# manually
-FetchContent_GetProperties(nlohmann-json)
-if (NOT nlohmann-json_POPULATED)
-    FetchContent_Populate(nlohmann-json)
-    include_directories(SYSTEM ${nlohmann-json_SOURCE_DIR}/single_include)
-endif ()
-
 
 ###############################
 # ANTLR CPP RUNTIME FOR THE SPARQL PARSER
 ###############################
+set(ANTLR_BUILD_CPP_TESTS OFF CACHE BOOL "don't try to build googletest twice")
+set(ANTLR_BUILD_SHARED OFF CACHE BOOL "We're only interested in the static library")
 FetchContent_Declare(
         antlr
         GIT_REPOSITORY https://github.com/antlr/antlr4.git
-        GIT_TAG 9239e6ff444420516b44b7621e8dc7691fcf0e16
+        GIT_TAG cc82115a4e7f53d71d9d905caa2c2dfa4da58899 # 4.13.12
+        SOURCE_SUBDIR runtime/Cpp
```

Comment:
(not related to anything)
We have a typo in the S2 Fetch content (GIT_TA vs GIT_TAG) which effectively doesn't pin the version...
Can you please also fix this here?